hatch_build.py

```diff
@@ -0,0 +1,83 @@
+from hatchling.builders.hooks.plugin.interface import BuildHookInterface
+from hatchling.metadata.plugin.interface import MetadataHookInterface
+
+import tomlkit
+import pathlib
+
+
+class PinVersionMetadataHook(MetadataHookInterface):
+    """Update the dependency versions so they are pinned metadata of the wheel artefact produced by a wheel."""
+    def update(self, metadata):
+        change_count = 0
+        # Overlay the pinned dependencies, replacing dependencies if they already exist.
+        for pinned_dep in self.config["config"]["pinned_dependencies"]:
+            pinned_dep_name = pinned_dep.split(" ")[0]
```

Comment:
I presume it's impossible for package names to have spaces and then I doubted myself, and thought I'd mention it.

Reply:
Per, [PEP 426](https://peps.python.org/pep-0426/#name) ( and PEP 508 ) spaces are not valid in a name, so this was the easiest way to get the package name for simple matching of the lists.

Reply:
Now I'm wondering if there are any unicode values which appear to be spaces but aren't really. But that wouldn't affect this code, thanks for the explanation, that's great :)